packages/nimble-components/src/table/models/keyboard-navigation-manager.ts

```diff
@@ -0,0 +1,1243 @@
+                    break;
+                case keyEnd:
+                    handled = this.onEndPressed(event.ctrlKey);
+                    break;
+                case keyEnter:
+                    handled = this.onEnterPressed(event.ctrlKey);
+                    break;
+                case keySpace:
+                    handled = this.onSpacePressed(event.shiftKey);
+                    break;
+                case keyFunction2:
+                    handled = this.onF2Pressed();
+                    break;
+                default:
+                    break;
+            }
+        }
+        if (handled) {
+            event.preventDefault();
+        }
+    };
+
+    private readonly onKeyDown = (event: KeyboardEvent): void => {
+        if (!this.inNavigationMode && !event.defaultPrevented) {
+            if (
+                event.key === keyEscape
+                && (this.focusType === TableFocusType.cellActionMenu
+                    || this.focusType === TableFocusType.cellContent)
+            ) {
+                const row = this.getCurrentRow();
+                if (row) {
+                    this.trySetCellFocus(row.getFocusableElements());
+                }
+            }
+        }
+    };
+
+    private readonly onViewportKeyDown = (event: KeyboardEvent): void => {
+        let handleEvent = !this.inNavigationMode
+            && (event.key === keyArrowUp || event.key === keyArrowDown);
+        switch (event.key) {
+            case keyPageUp:
+            case keyPageDown:
+            case keyHome:
+            case keyEnd:
+                handleEvent = true;
+                break;
+            default:
+                break;
+        }
+        if (handleEvent) {
+            // Swallow key presses that would cause table scrolling, independently of keyboard navigation
+            event.preventDefault();
+            event.stopImmediatePropagation();
+        }
+    };
+
+    private onEnterPressed(ctrlKey: boolean): boolean {
+        let row: TableRow | TableGroupRow | undefined;
+        let rowElements!: TableRowFocusableElements;
+        if (this.hasRowOrCellFocusType()) {
+            row = this.getCurrentRow();
+            rowElements = row!.getFocusableElements();
+        }
+        if (this.focusType === TableFocusType.row) {
+            if (row instanceof TableGroupRow) {
+                this.toggleRowExpanded(row);
+                return true;
+            }
+        }
+        if (this.focusType === TableFocusType.cell) {
+            if (ctrlKey) {
+                const cell = rowElements.cells[this.columnIndex]!;
+                if (cell.actionMenuButton && !cell.actionMenuButton.open) {
+                    cell.actionMenuButton.toggleButton!.control.click();
+                    return true;
+                }
+            }
+            return this.focusFirstInteractiveElementInCurrentCell(rowElements);
+        }
+        return false;
+    }
+
+    private onF2Pressed(): boolean {
```

Comment:
ARIA guidance also suggests that a subsequent press of `F2` restores navigation mode.

Reply:
F2 will now return to navigation mode (refocus the cell) from cellContent/cellActionMenu focus states, and added tests.